include/aspect/simulator_signals.h

```diff
@@ -80,7 +80,8 @@ namespace aspect
      * that can be modified and will be used to construct the
      * final finite element system later.
      */
-    boost::signals2::signal<void (std::vector<VariableDeclaration<dim> > &)>
+    boost::signals2::signal<void (const Parameters<dim> &parameters,
+                                  std::vector<VariableDeclaration<dim> > &)>
```

Comment:
That's the wrong way to go. If a function you want to call requires access to the parameters object, just pass a bound pointer to a function, e.g.
```
  edit_finite_element_variables.connect (std_cxx11::bind (&MyClass::my_function,
                                                                                            std_cxx11::cref(parameters),
                                                                                            std_cxx11::_1));
```
This requires that you have access to the parameters object where you call the `connect` function, but this way you won't have to adjust the argument list of signals all the time.